xbmc/cores/VideoPlayer/DVDInputStreams/DVDStateSerializer.cpp

```diff
@@ -189,105 +73,68 @@ bool CDVDStateSerializer::DVDToXMLState( std::string &xmlstate, const dvd_state_
+    std::string property = childElement->Value();
+    if (property == "title")
     {
-      text = TiXmlHandle( element ).FirstChildElement("value").FirstChild().Text();
-      if( text )
-        sscanf(text->Value(), "0x%hx", &state->registers.GPRM[index]);
-
-      text = TiXmlHandle( element ).FirstChildElement("mode").FirstChild().Text();
-      if( text )
-        sscanf(text->Value(), "0x%c", &state->registers.GPRM_mode[index]);
-
-      text = TiXmlHandle( element ).FirstChildElement("time").FirstChildElement("tv_sec").FirstChild().Text();
-      if( text )
-        sscanf(text->Value(), "%ld", &state->registers.GPRM_time[index].tv_sec);
-
-      text = TiXmlHandle( element ).FirstChildElement("time").FirstChildElement("tv_usec").FirstChild().Text();
-      if( text )
-        sscanf(text->Value(), "%ld", (long int*)&state->registers.GPRM_time[index].tv_usec);
+      state.title = static_cast<uint32_t>(std::stoul(childElement->GetText()));
     }
-    element = element->NextSiblingElement("gprm");
-  }
-
-  if( (text = hRoot.FirstChildElement("domain").FirstChild().Text()) )
-    sscanf(text->Value(), "%d", (int*) &state->domain);
-
-  if( (text = hRoot.FirstChildElement("vtsn").FirstChild().Text()) )
-    sscanf(text->Value(), "%d", &state->vtsN);
-
-  if( (text = hRoot.FirstChildElement("pgcn").FirstChild().Text()) )
-    sscanf(text->Value(), "%d", &state->pgcN);
-
-  if( (text = hRoot.FirstChildElement("pgn").FirstChild().Text()) )
-    sscanf(text->Value(), "%d", &state->pgN);
-
-  if( (text = hRoot.FirstChildElement("celln").FirstChild().Text()) )
-    sscanf(text->Value(), "%d", &state->cellN);
-
-  if( (text = hRoot.FirstChildElement("cell_restart").FirstChild().Text()) )
-    sscanf(text->Value(), "%d", &state->cell_restart);
-
-  if( (text = hRoot.FirstChildElement("blockn").FirstChild().Text()) )
-    sscanf(text->Value(), "%d", &state->blockN);
-
-  { TiXmlHandle hrsm = hRoot.FirstChildElement("rsm");
-
-    if( (text = hrsm.FirstChildElement("vtsn").FirstChild().Text()) )
-      sscanf(text->Value(), "%d", &state->rsm_vtsN);
-
-    if( (text = hrsm.FirstChildElement("blockn").FirstChild().Text()) )
-      sscanf(text->Value(), "%d", &state->rsm_blockN);
-
-    if( (text = hrsm.FirstChildElement("pgcn").FirstChild().Text()) )
-      sscanf(text->Value(), "%d", &state->rsm_pgcN);
-
-    if( (text = hrsm.FirstChildElement("celln").FirstChild().Text()) )
-      sscanf(text->Value(), "%d", &state->rsm_cellN);
-
-    element = hrsm.FirstChildElement("registers").FirstChildElement("sprm").Element();
-    while( element )
+    else if (property == "time")
     {
-      element->Attribute("index", &index);
-      text = TiXmlHandle(element).FirstChildElement("value").FirstChild().Text();
-      if( text && index >= 0 && index < 5 )
-        sscanf(text->Value(), "0x%hx", &state->rsm_regs[index]);
-
-      element = element->NextSiblingElement("sprm");
+      state.time = static_cast<int64_t>(std::stoul(childElement->GetText()));
```

Comment:
The std::stoul methods throw.you might want to handle that.

May also want to look into c++17 std::from_chars.

Reply:
thanks for the tip, seems to work just fine